lib/src/main/java/io/ably/lib/transport/Hosts.java

```diff
@@ -61,8 +61,15 @@ public Hosts(String primaryHost, String defaultHost, ClientOptions options) thro
         }
         fallbackHostsUseDefault = options.fallbackHostsUseDefault;
         if (options.fallbackHosts == null) {
-            fallbackHosts = Arrays.copyOf(Defaults.HOST_FALLBACKS, Defaults.HOST_FALLBACKS.length);
-            fallbackHostsIsDefault = true;
+            if (!fallbackHostsUseDefault && options.environment != null && !options.environment.equalsIgnoreCase("production")) {
```

Comment:
Also need to add check if port and tlsport are set in options
https://github.com/ably/ably-js/blob/8ba626804f68e262c81d9304fca9a05db1a5d7c6/common/lib/util/defaults.js#L138

Reply:
Also, https://github.com/ably/ably-ruby/pull/196#issue-378182770